essentials-framework/Essentials Devices Common/Essentials Devices Common/PartitionSensor/GlsPartitionSensorController.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Crestron.SimplSharp;
+using Crestron.SimplSharpPro;
+using Crestron.SimplSharpPro.DeviceSupport;
+using Crestron.SimplSharpPro.GeneralIO;
+using Newtonsoft.Json;
+using PepperDash.Core;
+using PepperDash.Essentials.Core;
+using PepperDash.Essentials.Core.Config;
+using PepperDash.Essentials.Core.Bridges;
+using PepperDash_Essentials_Core.Bridges.JoinMaps;
+
+namespace PepperDash.Essentials.Devices.Common.PartitionSensor
+{
+	[Description("Wrapper class for GLS Cresnet Partition Sensor")]
+	public class GlsPartitionSensorController : CrestronGenericBridgeableBaseDevice
+	{
+		public GlsPartCn PartitionSensor { get; private set; }
+
+		public StringFeedback NameFeedback { get; private set; }
+
+		public BoolFeedback EnableFeedback { get; private set; }
+
+		public BoolFeedback PartitionSensedFeedback { get; private set; }
+		public BoolFeedback PartitionNotSensedFeedback { get; private set; }
+
+		public IntFeedback SensitivityFeedback { get; private set; }
+
+		public bool InTestMode { get; private set; }
+		public bool TestEnableFeedback { get; private set; }
+		public bool TestPartitionSensedFeedback { get; private set; }
+		public int TestSensitivityFeedback { get; private set; }
+
+		public Func<string> NameFeedbackFunc

```

Comment:
I would recommend changing all these Funcs to be regular methods instead of public `Func<T>` properties. Easier to read and figure out what's going on.